### DISCUSSION TITLE
How should I do with`negative_sampling()` for bipartite graph

### DISCUSSION THREAD

**Discussion by ogawayuto** [pyg-team/pytorch_geometric#8323] [Q&A] [answered]
HI, I have a question about `torch_geometric.utils.negative_sampling()`. In the refrence you can give `(num_src_node, num_dst_node)` as `num_nodes` argument for bipartite graph. In the refrence you can see an example below.
(https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.negative_sampling)
 ```
edge_index = torch.as_tensor([[0, 0, 1, 2],
                              [0, 1, 2, 3]])
 negative_sampling(edge_index, num_nodes=(3, 4))
  >>> tensor([[0, 2, 2, 1],
         [2, 2, 1, 3]])
```

First, I assumed a graph with 3 nodes and a graph with 4 nodes and the function would return negative edges between two graphs. But, the example seems to return edges edges like self-loop, internal edges, edges between graphs. 

I'm confused with what the funcions do when you give bipartite graph. How should I do to get negative edges between two types of nodes in bipartite graph?

**Comment by akihironitta** [comment, marked as answer]
> the example seems to return edges edges like self-loop, internal edges, edges between graphs.

I believe the example is correct. It looks like a self-loop, but it's an edge from `2` (src node type) to `2` (dst node type).

**Reply by ogawayuto** [reply]
Thank you for your reply! I was confusing src indices and dst indices.
